pymobiledevice3/remote/core_device_tunnel_service.py

```diff
@@ -0,0 +1,472 @@
+
+        device_info = b'\xe7FaltIRK\x80\xe9\xe8-\xc0jIykVoT\x00\x19\xb1\xc7{FbtAddrQ14:98:77:76:76:EBCmacv\x14\x98wi' \
+                      b'\rq[remotepairing_serial_numberLC07HH0YHQ6NYIaccountIDa$3241983E-C373-4017-8963-0EB85C3CB8B7' \
+                      b'EmodelJMacmini9,1DnameQUser\xe2\x80\x99s Mac mini'
+
+        tlv = TLVBuf.build([
+            {'type': PairingDataComponentType.IDENTIFIER, 'data': self.identifier.encode()},
+            {'type': PairingDataComponentType.PUBLIC_KEY,
+             'data': self.ed25519_private_key.public_key().public_bytes_raw()},
+            {'type': PairingDataComponentType.SIGNATURE, 'data': self.signature},
+            {'type': PairingDataComponentType.INFO, 'data': device_info},
+        ])
+
+        cip = ChaCha20Poly1305(setup_encryption_key)
+        encrypted_data = cip.encrypt(b'\x00\x00\x00\x00PS-Msg05', tlv, b'')
+
+        tlv = TLVBuf.build([
+            {'type': PairingDataComponentType.ENCRYPTED_DATA, 'data': encrypted_data[:255]},
+            {'type': PairingDataComponentType.ENCRYPTED_DATA, 'data': encrypted_data[255:]},
+            {'type': PairingDataComponentType.STATE, 'data': b'\x05'},
+        ])
+
+        response = self._send_receive_pairing_data({
+            'data': tlv,
+            'kind': 'setupManualPairing',
+            'sendingHost': 'User’s Mac mini',
+            'startNewSession': False})
+        data = self.decode_tlv(TLVBuf.parse(response))
+
+        tlv = TLVBuf.parse(cip.decrypt(
+            b'\x00\x00\x00\x00PS-Msg06', data[PairingDataComponentType.ENCRYPTED_DATA], b''))
+
+        return tlv
+
+    def _init_client_server_main_encryption_keys(self) -> None:
+        client_key = HKDF(
+            algorithm=hashes.SHA512(),
+            length=32,
+            salt=None,
+            info=b'ClientEncrypt-main',
+        ).derive(self.encryption_key)
+        self.client_cip = ChaCha20Poly1305(client_key)
+
+        server_key = HKDF(
+            algorithm=hashes.SHA512(),
+            length=32,
+            salt=None,
+            info=b'ServerEncrypt-main',
+        ).derive(self.encryption_key)
+        self.server_cip = ChaCha20Poly1305(server_key)
+
+    def _create_remote_unlock(self) -> None:
+        response = self._send_receive_encrypted_request({'request': {'_0': {'createRemoteUnlockKey': {}}}})
+        self.remote_unlock_host_key = response['createRemoteUnlockKey']['hostKey']
+
+    def _create_listener(self, private_key: RSAPrivateKey, protocol: str = 'quic') -> Mapping:
+        request = {'request': {'_0': {'createListener': {
+            'key': base64.b64encode(
+                private_key.public_key().public_bytes(Encoding.DER, PublicFormat.SubjectPublicKeyInfo)
+            ).decode(),
+            'transportProtocolType': protocol}}}}
+
+        response = self._send_receive_encrypted_request(request)
+        return response['createListener']
+
+    def _attempt_validate_pairing(self) -> None:
```

Comment:
Why not attemp_pair_verify?

Reply:
- actually it was named that way in the first place, but i refactored it to be more similiar to `LockdownClient` design
- refactored back